ironic-deployment/basic-auth/default/auth.yaml

```
@@ -1,7 +1,7 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: capm3-ironic
+  name: ${NAMEPREFIX}-ironic
```

Do we have to make this configurable? I would much prefer to just skip the prefix. We don't have the prefix on the ConfigMaps and Secrets either.

If we absolutely need a prefix, I would suggest using the built-in `namePrefix` in [kustomize](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#setting-cross-cutting-fields). That way we don't need the custom bash/sed logic and the manifests stay clean.

```suggestion
  name: ironic
```

Same for all the other `capm3` -> `${NAMEPREFIX}` changes.

---

Yes, we need to have the name prefix configurable. The point is that Ironic needs to be in the same namespace as BMO. With CAPM3 v1a4, we have BMO deployed in CAPM3, so it is in `capm3-system` namespace. With v1a5, we plan to deploy BMO outside of CAPM3, so it is in `baremetal-operator-system` namespace. Therefore, we need to deploy Ironic in different namespace accordingly. However, the Ironic deployment is independent with BMO deployment. Therefore, `namePrefix` in kustomize doesn't help in this case. The only way is to use bash commands to check which CAPM3 we are using, and change the nameprefix accordingly.

---

Thanks for explaining! Ok so the namespace must be different, I do get that. But the Deployment name can always be the same, right? That would make this change smaller and simpler I think.

---

Yes, technically the name doesn't need to change. However, in v1a4, if we want to deploy Ironic in `capm3-system` namespace,  we should add the `capm3-` prefix since the others have the same prefix (we use namePrefix in CAPM3 kustomize). The same thing will be applied in v1a5 when BMO and Ironic are deployed in `baremetal-operator-system` namespace. That is why we also need to change the name for the sake of synchronization.